Do not load HPO-associated genes when loading OMIM terms

northwestwitch opened the pull request:
Fix #4151

Genes associated to disease are included in the min2genes.txt file, a file which we download from the HPO but we never use. It has likely more specific genes instead of the huge list obtained from the HPO terms

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>


**How to test**:
1. Load diseases using the latest files and make sure that the disease page contains the expected info

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by CR, DN
- [x] tests executed by CR

dnil commented:
Hm, I see no disease associated genes after 
```
scout update diseases -f .
```
![Screenshot 2023-10-20 at 14 26 21](https://github.com/Clinical-Genomics/scout/assets/758570/d322bbc0-17b9-41bc-a57b-ee28e70aae64)
Also fewer diseases??

northwestwitch commented:
I think that it shows what is found in min2genes, I've checked a few and those with no genes don't have genes in the file. The file looks like this in fact:

<img width="544" alt="image" src="https://github.com/Clinical-Genomics/scout/assets/28093618/351a5ed2-c000-40ae-a6f4-494b500f3148">

northwestwitch commented:
Discussing with @dnil I've realised that is probably genemap2 the file to use to do that, and that mim2genes perhaps was used in the past but has changed. We should remove it and the code parsing it!

northwestwitch commented:
This is ready for review now

dnil commented on scout/load/disease.py:
```diff
@@ -8,7 +8,7 @@
 from scout.parse.hpo_mappings import parse_hpo_annotations, parse_hpo_to_genes
 from scout.parse.omim import get_mim_phenotypes
-from scout.utils.scout_requests import fetch_hpo_disease_annotation, fetch_hpo_to_genes_to_disease
+from scout.utils.scout_requests import fetch_hpo_disease_annotation
```
👍

dnil commented on scout/server/blueprints/diagnoses/templates/diagnoses/omim_terms.html:
```diff
@@ -48,8 +47,13 @@
-                  <td><span class="text-body">{{ omim.hpo_terms|length }}</span></td>
+                  <td><span class="text-body">
+                    {% for term in omim.hpo_terms %}
+                      <a class="badge bg-secondary text-white" href="https://hpo.jax.org/app/browse/term/{{term}}" target="_blank" rel="noopener"  >{{term}}</a>
```
💎

dnil reviewed (commented):
Looks safe and simplifies things a bunch.

These "debug" warnings will stay with prod update:
`DEBUG The following gene symbols could not be found in database: {None}`
It's not essential, but it would make sense to either prune empty gene lists on `disease_term`, much like we do with other Pydantic-introduced empties. Or add a more appropriate `None`/empty-value check in `build_disease_term()`, around here https://github.com/Clinical-Genomics/scout/blob/e4a90c795521ef27c91597f462cedf9ee33e4f9d/scout/build/disease.py#L26.

northwestwitch commented:
> Looks safe and simplifies things a bunch.
> 
> These "debug" warnings will stay with prod update: `DEBUG The following gene symbols could not be found in database: {None}` It's not essential, but it would make sense to either prune empty gene lists on `disease_term`, much like we do with other Pydantic-introduced empties. Or add a more appropriate `None`/empty-value check in `build_disease_term()`, around here

I remember seeing those None before but I can't reproduce it any more? Tested both by updating  diseases from the latest files downloaded from OMIM and by pointing to the files in scout/demo/resources

dnil commented:
> > Looks safe and simplifies things a bunch.
> > These "debug" warnings will stay with prod update: `DEBUG The following gene symbols could not be found in database: {None}` It's not essential, but it would make sense to either prune empty gene lists on `disease_term`, much like we do with other Pydantic-introduced empties. Or add a more appropriate `None`/empty-value check in `build_disease_term()`, around here
> 
> I remember seeing those None before but I can't reproduce it any more? Tested both by updating diseases from the latest files downloaded from OMIM and by pointing to the files in scout/demo/resources

Well, all you need is to load with a release_4.72, then go from there. Possibly via downgrade to 4.71, which is what we will have on  `prod`. That just happened to also be the state of one of my locals.

northwestwitch commented:
> Well, all you need is to load with a release_4.72, then go from there. Possibly via downgrade to 4.71, which is what we will have on prod. That just happened to also be the state of one of my locals.

Switched to branch version_4.72 and loaded HPO/genes and disease, I see no Nones: (which is weird, I remember they were there)

<img width="1353" alt="image" src="https://github.com/Clinical-Genomics/scout/assets/28093618/b31b1fef-f86d-46f0-ad80-7325f74cf4f8">

Switched to branch version_4.71 and loaded HPO terms and OMIM terms, I don't see any Nones here as well:

<img width="1005" alt="image" src="https://github.com/Clinical-Genomics/scout/assets/28093618/4d17e79a-d104-4f48-a4f3-010202b5dfe6">

Switched to this branch and loaded OMIM terms: no Nones 🤔 

<img width="1023" alt="image" src="https://github.com/Clinical-Genomics/scout/assets/28093618/dab122e0-180d-478a-a756-9d3821ff0a64">

dnil commented on scout/build/disease.py:
```diff
@@ -24,6 +24,8 @@ def build_disease_term(disease_info: dict, alias_genes: dict = {}) -> dict:
     hgnc_symbols_not_found = set()
     hgnc_ids = set()
     for hgnc_symbol in disease_info.get("hgnc_symbols", []):
+        if hgnc_symbol is None:
```
```suggestion
        if None in hgnc_symbol:
```

northwestwitch replied:
That line should work, could you please check again? Before this change in the branch  I saw:

<img width="964" alt="image" src="https://github.com/Clinical-Genomics/scout/assets/28093618/361087b9-d968-441f-8aef-784222bf9a81">

After the change:

<img width="820" alt="image" src="https://github.com/Clinical-Genomics/scout/assets/28093618/2c9af073-c2b0-46e8-bae1-de9d22401399">

northwestwitch commented:
How the diseases pages look after loading diseases using this branch:

<img width="1439" alt="image" src="https://github.com/Clinical-Genomics/scout/assets/28093618/4af7ab90-1e92-415e-8dcc-2428f351f921">

dnil commented:
Nah, still the same:
![Screenshot 2023-10-23 at 11 54 11](https://github.com/Clinical-Genomics/scout/assets/758570/e711905e-d462-40e2-a2e7-dd9b25e87dec)

Thing is the value is not a `None`, I think it might be its a set with a `None`, but you're the Pydantic guru here!

northwestwitch commented:
> Thing is the value is not a `None`, I think it might be its a set with a `None`, but you're the Pydantic guru here!

The Pydantic validation hasn't even happened yet, That message is printed at line 41 of build_disease_term. Pydantic gets called at line 47.. I'm very confused now

northwestwitch commented:
Ok, I used branch 4.72 and I see no Nones

<img width="1155" alt="image" src="https://github.com/Clinical-Genomics/scout/assets/28093618/7911c2aa-feb4-4595-98e6-82eb2a8b1f03">

But at least I can reproduce the issue (the huge amount of genes):

<img width="703" alt="image" src="https://github.com/Clinical-Genomics/scout/assets/28093618/354e8a6e-2c4a-4913-aa42-127ddc68a19c">

<hr>

Then switch to 4.71 (after downgrading pydantic), and reload diseases, still no Nones

<img width="1147" alt="image" src="https://github.com/Clinical-Genomics/scout/assets/28093618/c49a6907-ee74-4c05-8cf7-5f7a73ffdd2b">

Database looks as expected, with no HPO genes (before the infamous bug fix that led to the memory sucking issue)

<img width="623" alt="image" src="https://github.com/Clinical-Genomics/scout/assets/28093618/364bb146-45e3-4ab6-832f-7ca6dc4ae2b5">

<ht>

Using this branch:

<img width="884" alt="image" src="https://github.com/Clinical-Genomics/scout/assets/28093618/b69b8913-4de8-47c4-ae7e-26a8f8173a9a">

Database analysis:

<img width="584" alt="image" src="https://github.com/Clinical-Genomics/scout/assets/28093618/29c25b0a-00a9-4f71-817a-643603938e59">

dnil reviewed (approved):
Retesting with 4.72 -> pydantic -> update diseases -> 4.71 -> pydantic -> update diseases -> this branch -> pydantic -> update disease is now nice and silent. Thank you!
